### PR TITLE
Remove `newID` function

### PR DESCRIPTION
**Pull request by dylanlott**
Removes `NewID` function from all tests and uses `provider.FullIdentity` instead.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -132,6 +132,7 @@ require (
 	github.com/prometheus/common v0.0.0-20180326160409-38c53a9f4bfc // indirect
 	github.com/prometheus/procfs v0.0.0-20180408092902-8b1c2da0d56d // indirect
 	github.com/segmentio/go-prompt v1.2.1-0.20161017233205-f0d19b6901ad // indirect
+	golang.org/x/tools v0.0.0-20181011204843-157aeed469d4 // indirect
```
Comment (egonelbre) [inline go.mod]: This shouldn't be here.

Comment (dylanlott) [reply go.mod]: Removed 👍

File: pkg/node/id.go [hunk 2]
```diff
@@ -46,20 +43,20 @@ func IDFromString(s string) *ID {
```
Comment (jenlij) [inline pkg/node/id.go]: can we remove this commented out code?

Comment (dylanlott) [reply pkg/node/id.go]: Ah, this must have been uncommented in a merge, I just removed it and pushed up. I'll make sure this build passes and then merge! 👍

**Pull request merged**